cypress/integration/visual.spec.ts

```diff
@@ -1,18 +1,46 @@
-const getCenterY = (el: HTMLElement) => {
-  const { y, height } = el.getBoundingClientRect();
+const getCenterY = ($el: JQuery<HTMLElement>) => {
+  const { y, height } = $el.get(0).getBoundingClientRect();
   return y + height / 2;
 };
 
+const isInViewport = ($el: JQuery<HTMLElement>) => {
+  const { fromElViewport } = Cypress.dom.getElementPositioning($el);
+  // Cypress types are broken for this at the moment
+  const win = Cypress.dom.getWindowByElement($el.get(0)) as any as Window;
+
+  return (
+    fromElViewport.top >= 0 &&
+    fromElViewport.right <= win.innerWidth &&
+    fromElViewport.bottom <= win.innerHeight &&
+    fromElViewport.left >= 0
+  );
+};
+
 describe('Visual', () => {
+  it('should have editor buttons in the viewport after the viewport gets downsized', () => {
+    cy.visit('/viz');
+
+    cy.getMonacoEditor();
+
+    cy.viewport(
+      Cypress.config().viewportWidth,
+      Cypress.config().viewportHeight - 300,
+    );
+
+    cy.findByRole('button', { name: 'Visualize' }).then(($button) => {
+      expect(isInViewport($button)).to.be.true;
+    });
```

Comment:
Question - can't we just `cy.click` this? Cypress should fail if the button isn't in the viewport

Reply:
No need to hold up the PR for this though, just curious

Reply:
Clicking doesn't work correctly - it automatically scrolls to the element (although it can be disabled with a flag, AFAIR) and focused the element (which makes the browser forcefully scroll to the element). The first isn't as much of a problem for us but still something that could blur the test scenario.

If I try to use `cy.click()` I end up with a state like this:
<img width="1131" alt="Screenshot 2021-08-30 at 18 41 43" src="https://user-images.githubusercontent.com/9800850/131374431-d125d0f0-b537-418d-927d-564d8b0cf613.png">

Note that this is without the fix being applied - the browser (most likely) has shifted the whole thing so the button became part of the viewport.